File: server-code/spring/instagram/src/main/java/org/openapitools/model/MediaEntry.java
package org.openapitools.model;

import java.net.URI;
import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import org.openapitools.model.CaptionData;
import org.openapitools.model.CommentsCollection;
import org.openapitools.model.ImagesData;
import org.openapitools.model.LikesCollection;
import org.openapitools.model.LocationInfo;
import org.openapitools.model.UserInPhoto;
import org.openapitools.model.UserShortInfo;
import org.openapitools.model.VideosData;
import org.openapitools.jackson.nullable.JsonNullable;
import java.time.OffsetDateTime;
import javax.validation.Valid;
import javax.validation.constraints.*;
import io.swagger.v3.oas.annotations.media.Schema;


import java.util.*;
import javax.annotation.Generated;

/**
 * MediaEntry
 */

@Generated(value = "org.openapitools.codegen.languages.SpringCodegen", date = "2024-01-06T13:51:07.987869-04:00[America/Lower_Princes]")
public class MediaEntry {

  private String attribution;

  private CaptionData caption;

  private CommentsCollection comments;

  private String createdTime;

  private String filter;

  private String id;

  private ImagesData images;

  private LikesCollection likes;

  private String link;

  private LocationInfo location;

  @Valid
  private List<String> tags;

  /**
   * Type of this media entry
   */
  public enum TypeEnum {
    IMAGE("image"),
    
    VIDEO("video");

    private String value;

    TypeEnum(String value) {
      this.value = value;
    }

    @JsonValue
    public String getValue() {
      return value;
    }

    @Override
    public String toString() {
      return String.valueOf(value);
    }

    @JsonCreator
    public static TypeEnum fromValue(String value) {
      for (TypeEnum b : TypeEnum.values()) {
        if (b.value.equals(value)) {
          return b;
        }
      }
      throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }
  }

  private TypeEnum type;

  private UserShortInfo user;

  private Boolean userHasLiked;

  @Valid
  private List<@Valid UserInPhoto> usersInPhoto;

  private VideosData videos;

  public MediaEntry attribution(String attribution) {
    this.attribution = attribution;
    return this;
  }

  /**
   * ??? Unknown ???
   * @return attribution
  */
  
  @Schema(name = "attribution", description = "??? Unknown ???", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("attribution")
  public String getAttribution() {
    return attribution;
  }

  public void setAttribution(String attribution) {
    this.attribution = attribution;
  }

  public MediaEntry caption(CaptionData caption) {
    this.caption = caption;
    return this;
  }

  /**
   * Get caption
   * @return caption
  */
  @Valid 
  @Schema(name = "caption", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("caption")
  public CaptionData getCaption() {
    return caption;
  }

  public void setCaption(CaptionData caption) {
    this.caption = caption;
  }

  public MediaEntry comments(CommentsCollection comments) {
    this.comments = comments;
    return this;
  }

  /**
   * Get comments
   * @return comments
  */
  @Valid 
  @Schema(name = "comments", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("comments")
  public CommentsCollection getComments() {
    return comments;
  }

  public void setComments(CommentsCollection comments) {
    this.comments = comments;
  }

  public MediaEntry createdTime(String createdTime) {
    this.createdTime = createdTime;
    return this;
  }

  /**
   * Media creation UNIX timestamp
   * @return createdTime
  */
  
  @Schema(name = "created_time", description = "Media creation UNIX timestamp", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("created_time")
  public String getCreatedTime() {
    return createdTime;
  }

  public void setCreatedTime(String createdTime) {
    this.createdTime = createdTime;
  }

  public MediaEntry filter(String filter) {
    this.filter = filter;
    return this;
  }

  /**
   * Filter of this media entry
   * @return filter
  */
  
  @Schema(name = "filter", description = "Filter of this media entry", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("filter")
  public String getFilter() {
    return filter;
  }

  public void setFilter(String filter) {
    this.filter = filter;
  }

  public MediaEntry id(String id) {
    this.id = id;
    return this;
  }

  /**
   * ID of a media entry
   * @return id
  */
  
  @Schema(name = "id", description = "ID of a media entry", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("id")
  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public MediaEntry images(ImagesData images) {
    this.images = images;
    return this;
  }

  /**
   * Get images
   * @return images
  */
  @Valid 
  @Schema(name = "images", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("images")
  public ImagesData getImages() {
    return images;
  }

  public void setImages(ImagesData images) {
    this.images = images;
  }

  public MediaEntry likes(LikesCollection likes) {
    this.likes = likes;
    return this;
  }

  /**
   * Get likes
   * @return likes
  */
  @Valid 
  @Schema(name = "likes", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("likes")
  public LikesCollection getLikes() {
    return likes;
  }

  public void setLikes(LikesCollection likes) {
    this.likes = likes;
  }

  public MediaEntry link(String link) {
    this.link = link;
    return this;
  }

  /**
   * Fixed URL of this media entry
   * @return link
  */
  
  @Schema(name = "link", description = "Fixed URL of this media entry", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("link")
  public String getLink() {
    return link;
  }

  public void setLink(String link) {
    this.link = link;
  }

  public MediaEntry location(LocationInfo location) {
    this.location = location;
    return this;
  }

  /**
   * Get location
   * @return location
  */
  @Valid 
  @Schema(name = "location", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("location")
  public LocationInfo getLocation() {
    return location;
  }

  public void setLocation(LocationInfo location) {
    this.location = location;
  }

  public MediaEntry tags(List<String> tags) {
    this.tags = tags;
    return this;
  }

  public MediaEntry addTagsItem(String tagsItem) {
    if (this.tags == null) {
      this.tags = new ArrayList<>();
    }
    this.tags.add(tagsItem);
    return this;
  }

  /**
   * List of tags assigned to this media
   * @return tags
  */
  
  @Schema(name = "tags", description = "List of tags assigned to this media", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("tags")
  public List<String> getTags() {
    return tags;
  }

  public void setTags(List<String> tags) {
    this.tags = tags;
  }

  public MediaEntry type(TypeEnum type) {
    this.type = type;
    return this;
  }

  /**
   * Type of this media entry
   * @return type
  */
  
  @Schema(name = "type", description = "Type of this media entry", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("type")
  public TypeEnum getType() {
    return type;
  }

  public void setType(TypeEnum type) {
    this.type = type;
  }

  public MediaEntry user(UserShortInfo user) {
    this.user = user;
    return this;
  }

  /**
   * Get user
   * @return user
  */
  @Valid 
  @Schema(name = "user", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("user")
  public UserShortInfo getUser() {
    return user;
  }

  public void setUser(UserShortInfo user) {
    this.user = user;
  }

  public MediaEntry userHasLiked(Boolean userHasLiked) {
    this.userHasLiked = userHasLiked;
    return this;
  }

  /**
   * Indicates whether authenticated user has liked this media or not
   * @return userHasLiked
  */
  
  @Schema(name = "user_has_liked", description = "Indicates whether authenticated user has liked this media or not", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("user_has_liked")
  public Boolean getUserHasLiked() {
    return userHasLiked;
  }

  public void setUserHasLiked(Boolean userHasLiked) {
    this.userHasLiked = userHasLiked;
  }

  public MediaEntry usersInPhoto(List<@Valid UserInPhoto> usersInPhoto) {
    this.usersInPhoto = usersInPhoto;
    return this;
  }

  public MediaEntry addUsersInPhotoItem(UserInPhoto usersInPhotoItem) {
    if (this.usersInPhoto == null) {
      this.usersInPhoto = new ArrayList<>();
    }
    this.usersInPhoto.add(usersInPhotoItem);
    return this;
  }

  /**
   * Users located on this media entry
   * @return usersInPhoto
  */
  @Valid 
  @Schema(name = "users_in_photo", description = "Users located on this media entry", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("users_in_photo")
  public List<@Valid UserInPhoto> getUsersInPhoto() {
    return usersInPhoto;
  }

  public void setUsersInPhoto(List<@Valid UserInPhoto> usersInPhoto) {
    this.usersInPhoto = usersInPhoto;
  }

  public MediaEntry videos(VideosData videos) {
    this.videos = videos;
    return this;
  }

  /**
   * Get videos
   * @return videos
  */
  @Valid 
  @Schema(name = "videos", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("videos")
  public VideosData getVideos() {
    return videos;
  }

  public void setVideos(VideosData videos) {
    this.videos = videos;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    MediaEntry mediaEntry = (MediaEntry) o;
    return Objects.equals(this.attribution, mediaEntry.attribution) &&
        Objects.equals(this.caption, mediaEntry.caption) &&
        Objects.equals(this.comments, mediaEntry.comments) &&
        Objects.equals(this.createdTime, mediaEntry.createdTime) &&
        Objects.equals(this.filter, mediaEntry.filter) &&
        Objects.equals(this.id, mediaEntry.id) &&
        Objects.equals(this.images, mediaEntry.images) &&
        Objects.equals(this.likes, mediaEntry.likes) &&
        Objects.equals(this.link, mediaEntry.link) &&
        Objects.equals(this.location, mediaEntry.location) &&
        Objects.equals(this.tags, mediaEntry.tags) &&
        Objects.equals(this.type, mediaEntry.type) &&
        Objects.equals(this.user, mediaEntry.user) &&
        Objects.equals(this.userHasLiked, mediaEntry.userHasLiked) &&
        Objects.equals(this.usersInPhoto, mediaEntry.usersInPhoto) &&
        Objects.equals(this.videos, mediaEntry.videos);
  }

  @Override
  public int hashCode() {
    return Objects.hash(attribution, caption, comments, createdTime, filter, id, images, likes, link, location, tags, type, user, userHasLiked, usersInPhoto, videos);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class MediaEntry {\n");
    sb.append("    attribution: ").append(toIndentedString(attribution)).append("\n");
    sb.append("    caption: ").append(toIndentedString(caption)).append("\n");
    sb.append("    comments: ").append(toIndentedString(comments)).append("\n");
    sb.append("    createdTime: ").append(toIndentedString(createdTime)).append("\n");
    sb.append("    filter: ").append(toIndentedString(filter)).append("\n");
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    images: ").append(toIndentedString(images)).append("\n");
    sb.append("    likes: ").append(toIndentedString(likes)).append("\n");
    sb.append("    link: ").append(toIndentedString(link)).append("\n");
    sb.append("    location: ").append(toIndentedString(location)).append("\n");
    sb.append("    tags: ").append(toIndentedString(tags)).append("\n");
    sb.append("    type: ").append(toIndentedString(type)).append("\n");
    sb.append("    user: ").append(toIndentedString(user)).append("\n");
    sb.append("    userHasLiked: ").append(toIndentedString(userHasLiked)).append("\n");
    sb.append("    usersInPhoto: ").append(toIndentedString(usersInPhoto)).append("\n");
    sb.append("    videos: ").append(toIndentedString(videos)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: server-code/aspnetcore/instagram/src/Org.OpenAPITools/Models/MediaEntry.cs
/*
 * Instagram API
 *
 * Description of Instagram RESTful API.  Current limitations:   * Instagram service does not support [cross origin headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS)   for security reasons, therefore it is not possible to use Swagger UI and make API calls directly from browser.   * Modification API requests (`POST`, `DELETE`) require additional security [scopes](https://instagram.com/developer/authorization/)   that are available for Apps [created on or after Nov 17, 2015](http://instagram.com/developer/review/) and   started in [Sandbox Mode](http://instagram.com/developer/sandbox/).   * Consider the [Instagram limitations](https://instagram.com/developer/limits/) for API calls that depends on App Mode.  **Warning:** For Apps [created on or after Nov 17, 2015](http://instagram.com/developer/changelog/) API responses containing media objects no longer return the `data` field in `comments` and `likes` nodes.  Last update: 2015-11-28 
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 * Generated by: https://openapi-generator.tech
 */

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Org.OpenAPITools.Converters;

namespace Org.OpenAPITools.Models
{ 
    /// <summary>
    /// 
    /// </summary>
    [DataContract]
    public partial class MediaEntry : IEquatable<MediaEntry>
    {
        /// <summary>
        /// ??? Unknown ???
        /// </summary>
        /// <value>??? Unknown ???</value>
        [DataMember(Name="attribution", EmitDefaultValue=false)]
        public string Attribution { get; set; }

        /// <summary>
        /// Gets or Sets Caption
        /// </summary>
        [DataMember(Name="caption", EmitDefaultValue=false)]
        public CaptionData Caption { get; set; }

        /// <summary>
        /// Gets or Sets Comments
        /// </summary>
        [DataMember(Name="comments", EmitDefaultValue=false)]
        public CommentsCollection Comments { get; set; }

        /// <summary>
        /// Media creation UNIX timestamp
        /// </summary>
        /// <value>Media creation UNIX timestamp</value>
        [DataMember(Name="created_time", EmitDefaultValue=false)]
        public string CreatedTime { get; set; }

        /// <summary>
        /// Filter of this media entry
        /// </summary>
        /// <value>Filter of this media entry</value>
        [DataMember(Name="filter", EmitDefaultValue=false)]
        public string Filter { get; set; }

        /// <summary>
        /// ID of a media entry
        /// </summary>
        /// <value>ID of a media entry</value>
        [DataMember(Name="id", EmitDefaultValue=false)]
        public string Id { get; set; }

        /// <summary>
        /// Gets or Sets Images
        /// </summary>
        [DataMember(Name="images", EmitDefaultValue=false)]
        public ImagesData Images { get; set; }

        /// <summary>
        /// Gets or Sets Likes
        /// </summary>
        [DataMember(Name="likes", EmitDefaultValue=false)]
        public LikesCollection Likes { get; set; }

        /// <summary>
        /// Fixed URL of this media entry
        /// </summary>
        /// <value>Fixed URL of this media entry</value>
        [DataMember(Name="link", EmitDefaultValue=false)]
        public string Link { get; set; }

        /// <summary>
        /// Gets or Sets Location
        /// </summary>
        [DataMember(Name="location", EmitDefaultValue=false)]
        public LocationInfo Location { get; set; }

        /// <summary>
        /// List of tags assigned to this media
        /// </summary>
        /// <value>List of tags assigned to this media</value>
        [DataMember(Name="tags", EmitDefaultValue=false)]
        public List<string> Tags { get; set; }


        /// <summary>
        /// Type of this media entry
        /// </summary>
        /// <value>Type of this media entry</value>
        [TypeConverter(typeof(CustomEnumConverter<TypeEnum>))]
        [JsonConverter(typeof(Newtonsoft.Json.Converters.StringEnumConverter))]
        public enum TypeEnum
        {
            
            /// <summary>
            /// Enum ImageEnum for image
            /// </summary>
            [EnumMember(Value = "image")]
            ImageEnum = 1,
            
            /// <summary>
            /// Enum VideoEnum for video
            /// </summary>
            [EnumMember(Value = "video")]
            VideoEnum = 2
        }

        /// <summary>
        /// Type of this media entry
        /// </summary>
        /// <value>Type of this media entry</value>
        [DataMember(Name="type", EmitDefaultValue=true)]
        public TypeEnum Type { get; set; }

        /// <summary>
        /// Gets or Sets User
        /// </summary>
        [DataMember(Name="user", EmitDefaultValue=false)]
        public UserShortInfo User { get; set; }

        /// <summary>
        /// Indicates whether authenticated user has liked this media or not
        /// </summary>
        /// <value>Indicates whether authenticated user has liked this media or not</value>
        [DataMember(Name="user_has_liked", EmitDefaultValue=true)]
        public bool UserHasLiked { get; set; }

        /// <summary>
        /// Users located on this media entry
        /// </summary>
        /// <value>Users located on this media entry</value>
        [DataMember(Name="users_in_photo", EmitDefaultValue=false)]
        public List<UserInPhoto> UsersInPhoto { get; set; }

        /// <summary>
        /// Gets or Sets Videos
        /// </summary>
        [DataMember(Name="videos", EmitDefaultValue=false)]
        public VideosData Videos { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class MediaEntry {\n");
            sb.Append("  Attribution: ").Append(Attribution).Append("\n");
            sb.Append("  Caption: ").Append(Caption).Append("\n");
            sb.Append("  Comments: ").Append(Comments).Append("\n");
            sb.Append("  CreatedTime: ").Append(CreatedTime).Append("\n");
            sb.Append("  Filter: ").Append(Filter).Append("\n");
            sb.Append("  Id: ").Append(Id).Append("\n");
            sb.Append("  Images: ").Append(Images).Append("\n");
            sb.Append("  Likes: ").Append(Likes).Append("\n");
            sb.Append("  Link: ").Append(Link).Append("\n");
            sb.Append("  Location: ").Append(Location).Append("\n");
            sb.Append("  Tags: ").Append(Tags).Append("\n");
            sb.Append("  Type: ").Append(Type).Append("\n");
            sb.Append("  User: ").Append(User).Append("\n");
            sb.Append("  UserHasLiked: ").Append(UserHasLiked).Append("\n");
            sb.Append("  UsersInPhoto: ").Append(UsersInPhoto).Append("\n");
            sb.Append("  Videos: ").Append(Videos).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }

        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="obj">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object obj)
        {
            if (obj is null) return false;
            if (ReferenceEquals(this, obj)) return true;
            return obj.GetType() == GetType() && Equals((MediaEntry)obj);
        }

        /// <summary>
        /// Returns true if MediaEntry instances are equal
        /// </summary>
        /// <param name="other">Instance of MediaEntry to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(MediaEntry other)
        {
            if (other is null) return false;
            if (ReferenceEquals(this, other)) return true;

            return 
                (
                    Attribution == other.Attribution ||
                    Attribution != null &&
                    Attribution.Equals(other.Attribution)
                ) && 
                (
                    Caption == other.Caption ||
                    Caption != null &&
                    Caption.Equals(other.Caption)
                ) && 
                (
                    Comments == other.Comments ||
                    Comments != null &&
                    Comments.Equals(other.Comments)
                ) && 
                (
                    CreatedTime == other.CreatedTime ||
                    CreatedTime != null &&
                    CreatedTime.Equals(other.CreatedTime)
                ) && 
                (
                    Filter == other.Filter ||
                    Filter != null &&
                    Filter.Equals(other.Filter)
                ) && 
                (
                    Id == other.Id ||
                    Id != null &&
                    Id.Equals(other.Id)
                ) && 
                (
                    Images == other.Images ||
                    Images != null &&
                    Images.Equals(other.Images)
                ) && 
                (
                    Likes == other.Likes ||
                    Likes != null &&
                    Likes.Equals(other.Likes)
                ) && 
                (
                    Link == other.Link ||
                    Link != null &&
                    Link.Equals(other.Link)
                ) && 
                (
                    Location == other.Location ||
                    Location != null &&
                    Location.Equals(other.Location)
                ) && 
                (
                    Tags == other.Tags ||
                    Tags != null &&
                    other.Tags != null &&
                    Tags.SequenceEqual(other.Tags)
                ) && 
                (
                    Type == other.Type ||
                    
                    Type.Equals(other.Type)
                ) && 
                (
                    User == other.User ||
                    User != null &&
                    User.Equals(other.User)
                ) && 
                (
                    UserHasLiked == other.UserHasLiked ||
                    
                    UserHasLiked.Equals(other.UserHasLiked)
                ) && 
                (
                    UsersInPhoto == other.UsersInPhoto ||
                    UsersInPhoto != null &&
                    other.UsersInPhoto != null &&
                    UsersInPhoto.SequenceEqual(other.UsersInPhoto)
                ) && 
                (
                    Videos == other.Videos ||
                    Videos != null &&
                    Videos.Equals(other.Videos)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                var hashCode = 41;
                // Suitable nullity checks etc, of course :)
                    if (Attribution != null)
                    hashCode = hashCode * 59 + Attribution.GetHashCode();
                    if (Caption != null)
                    hashCode = hashCode * 59 + Caption.GetHashCode();
                    if (Comments != null)
                    hashCode = hashCode * 59 + Comments.GetHashCode();
                    if (CreatedTime != null)
                    hashCode = hashCode * 59 + CreatedTime.GetHashCode();
                    if (Filter != null)
                    hashCode = hashCode * 59 + Filter.GetHashCode();
                    if (Id != null)
                    hashCode = hashCode * 59 + Id.GetHashCode();
                    if (Images != null)
                    hashCode = hashCode * 59 + Images.GetHashCode();
                    if (Likes != null)
                    hashCode = hashCode * 59 + Likes.GetHashCode();
                    if (Link != null)
                    hashCode = hashCode * 59 + Link.GetHashCode();
                    if (Location != null)
                    hashCode = hashCode * 59 + Location.GetHashCode();
                    if (Tags != null)
                    hashCode = hashCode * 59 + Tags.GetHashCode();
                    
                    hashCode = hashCode * 59 + Type.GetHashCode();
                    if (User != null)
                    hashCode = hashCode * 59 + User.GetHashCode();
                    
                    hashCode = hashCode * 59 + UserHasLiked.GetHashCode();
                    if (UsersInPhoto != null)
                    hashCode = hashCode * 59 + UsersInPhoto.GetHashCode();
                    if (Videos != null)
                    hashCode = hashCode * 59 + Videos.GetHashCode();
                return hashCode;
            }
        }

        #region Operators
        #pragma warning disable 1591

        public static bool operator ==(MediaEntry left, MediaEntry right)
        {
            return Equals(left, right);
        }

        public static bool operator !=(MediaEntry left, MediaEntry right)
        {
            return !Equals(left, right);
        }

        #pragma warning restore 1591
        #endregion Operators
    }
}
